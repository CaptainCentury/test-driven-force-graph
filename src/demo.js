import React from "react";
import styled from "styled-components";

import ForceGraph from "./components/force-graph";
import namesData from "./data/names";
import abcData from "./data/abc";

const BasicButton = styled.button`
  background: blue;
  border-radius: 3px;
  border: none;
  color: white;
  margin: 2px;
`;

export const DataButton = styled(BasicButton)`
  background: green;
`;

export const Demo = () => {
  const [radius, setRadius] = React.useState(10);
  const [linkStrength, setLinkStrength] = React.useState(0.01);
  const [data, setData] = React.useState(namesData);
  const [visualizer, setVisualizer] = React.useState(
    ForceGraph.defaultProps.visualizer
  );

  React.useEffect(() => {
    const update = visualizer.instance();
    update.radius = radius;
    update.linkStrength = linkStrength;
    setVisualizer(update);
  }, [linkStrength, radius]);

  return (
    <>
      <ForceGraph
        style={{ color: "blue" }}
        visualizer={visualizer}
        dataset={data}
        labelMode="labels"
      >
        <div>
          <DataButton
            onClick={() => {
              setData(namesData);
            }}
          >
            Names
          </DataButton>
          <DataButton
            onClick={() => {
              setData(abcData);
            }}
          >
            ABC
          </DataButton>
        </div>
      </ForceGraph>
      <p>
        Node radius:
        <input
          type="number"
          step="1"
          value={radius}
          id="radius"
          min="5"
          onChange={(event) => {
            setRadius(event.target.value);
          }}
        />
      </p>
      <p>
        Bond strength:
        <input
          type="number"
          step="0.01"
          value={Number(linkStrength)}
          id="link-strength"
          min="0.001"
          onChange={(event) => {
            setLinkStrength(event.target.value);
          }}
        />
      </p>
    </>
  );
};
