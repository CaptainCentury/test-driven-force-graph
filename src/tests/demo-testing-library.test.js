/**
 * @jest-environment jsdom
 */
import React from "react";
import { render, screen } from "@testing-library/react";

import { Demo } from "../demo";

describe("Demo component", () => {
  beforeEach(() => {
    render(<Demo />);
  });

  it("renders Abc dataset button", () => {
    const element = screen.getByText("ABC");
    expect(element.tagName).toBe("BUTTON");
  });

  it("renders Names dataset button", () => {
    const element = screen.getByText("Names");
    expect(element.tagName).toBe("BUTTON");
  });

  it("renders correct number of buttons", () => {
    const elements = screen.getAllByRole("button");
    expect(elements.length).toBe(2);
  });
});
