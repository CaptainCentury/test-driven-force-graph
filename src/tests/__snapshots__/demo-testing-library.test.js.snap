// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data button to render correctly 1`] = `
.c0 {
  background: blue;
  border-radius: 3px;
  border: none;
  color: white;
  margin: 2px;
}

.c1 {
  background: green;
}

<button
  class="c0 c1"
>
  Test
</button>
`;
